doc: release: 3.7: Add Flash Map release notes (#75045)
zephyrproject-rtos/zephyr: doc/releases/release-notes-3.7.rst

  * FS: It is now possible to truncate a file while opening using :c:func:`fs_open`
    and by passing ``FS_O_TRUNC`` flag.

  * Flash Map: TinyCrypt has been replaced with PSA Crypto in Flash Area integrity check.

>   * Flash Map: :c:func:`flash_area_flatten` has been added to be used where erase

Reviewer: ```suggestion
  * Flash Map: :c:func:`flash_area_flatten` has been added to be used where an erase
```

    operation has been previously used for removing/scrambling data rather than

>     to prepare device for random data write.

Reviewer: ```suggestion
    to prepare a device for a random data write.
```

  * Flash Map: :c:macro:`FIXED_PARTITION_NODE_OFFSET`, :c:macro:`FIXED_PARTITION_NODE_SIZE`
    and :c:macro:`FIXED_PARTITION_NODE_DEVICE` have been added to allow obtaining

>     fixed partition information by devicetree node rather than label.

Reviewer: ```suggestion
    fixed partition information from a devicetree node rather than a label.
```
maybe better?
